binaries/coordinator/src/lib.rs

```diff
@@ -497,6 +521,44 @@ async fn reload_dataflow(
     Ok(())
 }
 
+async fn retrieve_logs(
+    running_dataflows: &HashMap<Uuid, RunningDataflow>,
+    dataflow_id: Uuid,
+    node_id: NodeId,
+    daemon_connections: &mut HashMap<String, TcpStream>,
+) -> eyre::Result<Vec<u8>> {
+    let Some(dataflow) = running_dataflows.get(&dataflow_id) else {
+        bail!("No running dataflow found with UUID `{dataflow_id}`")
+    };
+    let message = serde_json::to_vec(&DaemonCoordinatorEvent::Logs {
+        dataflow_id,
+        node_id,
+    })?;
+    let mut reply_logs = Vec::new();
+    for machine_id in &dataflow.machines {
+        let daemon_connection = daemon_connections
+            .get_mut(machine_id)
+            .wrap_err("no daemon connection")?; // TODO: take from dataflow spec
+        tcp_send(daemon_connection, &message)
+            .await
+            .wrap_err("failed to send reload message to daemon")?;
+
+        // wait for reply
+        let reply_raw = tcp_receive(daemon_connection)
+            .await
+            .wrap_err("failed to receive reload reply from daemon")?;
+        match serde_json::from_slice(&reply_raw)
+            .wrap_err("failed to deserialize reload reply from daemon")?
+        {
+            DaemonCoordinatorReply::Logs { logs } => reply_logs = logs,
```

Comment:
If the dataflow is split across multiple machines, this always returns the logs from the last one, which might not be running the node.

Reply:
Agreed. I will rebase on top of #256 once merged and retrieve the Description of the dataflow to target the right machine.

Reply:
I have rebased on top of multiple daemon and reused resolved node description to retrieve the deployment of the node and only query the corresponding daemon in [541857a](https://github.com/dora-rs/dora/pull/259/commits/541857aed0b3551e8daf1449a1d75946d1aa4b4c)